compat/winansi.c

```diff
@@ -594,6 +594,35 @@ static void detect_msys_tty(int fd)
 
 #endif
 
+static HANDLE stdout_handle;
+static DWORD stdout_console_mode;
+
+static void reset_stdout_console_mode(void)
+{
+	SetConsoleMode(stdout_handle, stdout_console_mode);
+}
+
+static int enable_virtual_processing(void)
+{
+	stdout_handle = GetStdHandle(STD_OUTPUT_HANDLE);
```

Comment:
We could theoretically encounter situations where `stdout` isn't a console, but `stderr`is, but it's probably fine to not use native ANSI processing in that edge case.

Reply:
Good point. In such a scenario, we would still want to use virtual processing (the `winansi_thread` is supposed to only handle console handles, so `stdout` would not need that thread to run, either).

Let me think about the best way to resolve this.